src/devices/device.ts

```diff
@@ -13,29 +18,48 @@ export class Device extends EventEmitter {
     public autoSubscribe: boolean = true;
     public values: {[event: string]: any} = {};
 
+    protected _modeCount: number = 0;
+    protected _modes: IMode[] = [];
     protected _mode: number | undefined;
+    protected _modeMap: {[event: string]: number} = {};
```

Comment:
`_modeMap` is now protected so it can be used in device class to subscribe to mode